packages/server/src/modules/school/school.resolver.ts

```diff
@@ -1,21 +1,31 @@
 import { Args, Query, Resolver } from "@nestjs/graphql";
 import { School, SchoolCourse } from "src/@generated";
-import { SchoolCoursesArgs } from "src/modules/school/school.args";
+import {
+  SchoolCoursesArgs,
+  SchoolQueryPayload,
+} from "src/modules/school/school.args";
 import { PrismaService } from "../prisma/prisma.service";
 
 @Resolver()
 export class SchoolResolver {
   constructor(private readonly prisma: PrismaService) {}
 
   @Query(() => [School])
-  async schools() {
-    return await this.prisma.school.findMany({});
+  async schools(@Args() { retailLocationId }: SchoolQueryPayload) {
+    return await this.prisma.school.findMany({
+      where: {
+        provinceCode: retailLocationId.toLocaleUpperCase(),
```

Comment:
Here I suggest you to use `mode: 'insensitive'`. 
Look for where that is used inside this codebase, there are some spots where we used something like that.